fairseq/training.py

```diff
@@ -170,6 +177,12 @@ def __init__(
         kwargs = locals()
         del kwargs["self"]
 
+        # save start checkpoint and rename to checkpoint_latest
+        self.start_checkpoint = None
+        if fairseq_hydra_config.data.get("checkpoint", {}).get("restore_file") is not None:
+            self.start_checkpoint = fairseq_hydra_config.data["checkpoint"]["restore_file"]
+            fairseq_hydra_config.data["checkpoint"]["restore_file"] = "checkpoint_last.pt"
```

Comment:
You said the default value is already `"checkpoint_last.pt"` right? then we could simplify to
```suggestion
        self.start_checkpoint = fairseq_hydra_config.data.get("checkpoint", {}).pop("restore_file")
```
(pop removes the key/value pair and returns None if they key does not exist.)

Reply:
Yep, good point!